File: Interpolation and Fitting/NewtonInterpolation.jl
using PyPlot
function diff(x::Array,y::Array)
    m=length(x)
    a=Array{Float64}(undef,m)
    for i in 1:m
        a[i]=y[i]
    end
    for j in 2:m
        for i in reverse(collect(j:m))
            a[i]=(a[i]-a[i-1])/(x[i]-x[i-(j-1)])
        end
    end
    return a
end
function newton(x::Array,y::Array,z)
    m=length(x)
    a=diff(x,y)
    sum,pr=a[1],1.0
    for j in 1:(m-1)
        pr=pr*(z-x[j])
        sum=sum+a[j+1]*pr
    end
    return sum
end
function Poltt(x::Array,y::Array)
    cla()
    M,m=max(x...),min(x...)
    t=(M-m)/10
    xaxis=m-t:1/1000:M+t
    yvals=map(t -> newton(x,y,t),xaxis)
    display(length(xaxis))
    plot(xaxis,yvals)
    scatter(x,y)
    gcf()
end
Poltt([1.765,1.760,1.755,1.750],[0.92256,0.92137,0.92021,0.91906])

File: Interpolation and Fitting/NewtonInterpolation.py
import numpy as np
import matplotlib.pyplot as plt
def Newton(A,*b):
    def diff(A):
        A=np.array(A,dtype=float)
        n=A.shape[0]
        a=[A[i,1] for i in range(n)]
        for j in range(1,n):
            for i in range(n-1,j-1,-1):
                a[i]=(a[i]-a[i-1])/(A[i,0]-A[i-j,0])
        return a
    def newton(A,x):
        m=A.shape[0]
        a=diff(A)
        sum,pr=a[0],1.0
        for i in range(1,m):
            pr=pr*(x-A[i-1,0])
            sum=sum+a[i]*pr
        return sum
    A=np.array(A,dtype=float)
    if len(b)!=3:
        M,m=np.max(A[:,0]),np.min(A[:,0])
        xis=np.linspace(m,M,round((M-m)*1000))
    else:
        xis=np.linspace(b[1],b[2],round((b[2]-b[1])*1000))
    yis=[newton(A,xis[i]) for i in range(len(xis))]
    if len(b)!=0:
        print('F({})={}'.format(b[0],newton(A,b[0])))
    plt.scatter(A[:,0],A[:,1],color="b")
    plt.plot(xis,yis,color="r");plt.show()
A=np.array([[0.4,0.41075],[0.55,0.57815],[0.65,0.69675],[0.80,0.90],[0.95,1],[1.05,1.25382]])
Newton(A,0.99)
